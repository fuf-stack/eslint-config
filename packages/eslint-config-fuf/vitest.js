// TODO: migrate to v0.5.0 + (eslint 9)
// see: https://github.com/veritem/eslint-plugin-vitest

module.exports = {
  plugins: ['vitest'],
  overrides: [
    // test file rules
    {
      files: ['*.spec.ts', '*.spec.tsx', '*.test.ts', '*.test.tsx'],
      extends: ['plugin:vitest/recommended'],
      rules: {
        // Do not enforce production dependencies in tests
        'import/no-extraneous-dependencies': 'off',
      },
    },
  ],
};
