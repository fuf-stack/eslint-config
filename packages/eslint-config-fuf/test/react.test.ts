import { expect, it } from 'vitest';

import { lintFixture } from './helper';

it('enforces component arrow-function definition', async () => {
  const { fixedContent } = await lintFixture(
    'react-function-component-definition.tsx',
    'react.config.js',
  );
  expect(fixedContent).toMatchSnapshot();
});
