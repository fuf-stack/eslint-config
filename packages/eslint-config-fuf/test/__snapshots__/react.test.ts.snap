// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`enforces component arrow-function definition 1`] = `
"const Component = () => {
  return <div />;
};

export default Component;
"
`;
